### PR TITLE
inline=X ⇒ usePortal=!X

### LINKED ISSUE #1341: Proposal: rename inline=false to portal=true
What is "inline" really? This name has been confusing since the beginning, and we often talk about non-inlined Popovers, etc. 

Implementationally, `inline={false}` means "render in a `Portal` to remove from DOM flow."

I therefore propose a new name for the prop: `portal={true}`. This makes the semantics obvious. Enabling `portal` will render the children inside a `Portal`; disabling will not use a `Portal` (preserving DOM flow). Boom done.

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- `inline=X` &rArr; `usePortal=!X`
  - `Overlay/Dialog` default `usePortal=true`
  - 🔥 `Popover/Tooltip` default `usePortal=false` (cuz Popper.js is really great)
- rewrite docs about this prop

### CODE REVIEW

**Comment by adidahiya** [comment]
> Popover/Tooltip default usePortal=false (cuz Popper.js is really great)

please elaborate on this.

**Comment by giladgray** [comment]
I have found that _in simple cases_ inline Popovers (powered by Popper.js) work identically to portaled popovers. This was _not_ the case with Tether, but "Popper.js is really great," so things just work now.

I also found that _in most cases_ I prefer to use inline Popovers cuz they're easier to style and reason about. So with the improved functionality from Popper.js, seems reasonable to change the default.

**Comment by adidahiya** [comment]
How does it work? Is Popper creating its own portal-like thing under the hood?

**Comment by giladgray** [comment]
no, Popper is just better at positioning than Tether. 
`usePortal` is really only necessary if you have `overflow: hidden` on an ancestor but you want the popover to be able to overflow that ancestor. classic example is the table row use case.

**Comment by giladgray** [comment]
@adidahiya play with this example for a bit, look at the inline styles that Popper injects in each case, and then get back to me: http://localhost:9000/#core/components/popover.inline-rendering

File: packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ export class CollapsibleListExample extends BaseExample<ICollapsibleListExampleS
                 <RadioGroup
                     key="collapseFrom"
                     name="collapseFrom"
+                    inline={true}
```
Comment (adidahiya) [inline packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx]: what's this?

Comment (giladgray) [reply packages/docs-app/src/examples/core-examples/collapsibleListExample.tsx]: the _other_ inline modifier, the one that makes checkboxes etc `display: inline-block` so they'll lie on the same line

**Comment by giladgray** [comment]
#2053 will fix the test failures here because it removes those tests (MenuItem dynamic layout)!

**Pull request merged**